### PR TITLE
Added new "procure_method" which made fix to "mts_else_mto"

### PR DESCRIPTION
**Pull request by rrakso**
I've added the next procure method, because that orginal one "mts_else_mto" didn't work as we needed.
### Here is [more info](https://github.com/OCA/stock-logistics-warehouse/issues/727#issuecomment-570613107)

**Description of the issue/feature this PR addresses:**
The Supply Method on stock rule

**Current behavior before PR:**
If you have 5 pieces on stock and customer want 12, the system will _not take_ 5 from stock and manufacture remaining 7

**Desired behavior after PR is merged:**
If you have 5 pieces on stock and customer want 12, the system will _take_ 5 from stock and manufacture remaining 7.

After the PR is merged in Odoo will show new supply method:
image

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_rule.py [hunk 1]
```diff
@@ -39,31 +39,39 @@ class StockRule(models.Model):
...
+    location_src_id = fields.Many2one(
+        'stock.location', 'Source Location', check_company=True)
+    route_id = fields.Many2one(
+        'stock.location.route', 'Route', required=True, ondelete='cascade')
     procure_method = fields.Selection([
         ('make_to_stock', 'Take From Stock'),
         ('make_to_order', 'Trigger Another Rule'),
-        ('mts_else_mto', 'Take From Stock, if unavailable, Trigger Another Rule')], string='Supply Method', default='make_to_stock', required=True,
+        ('mts_else_mto', 'Take From Stock, if unavailable, Trigger Another Rule'),
+        ('mts_else_mto_ByERS', 'MTS+MTO by ERS')], string='Supply Method', default='make_to_stock', required=True,
```
Comment (rrakso) [inline addons/stock/models/stock_rule.py]: Here I've added new method

**Comment by pedrobaeza** [comment]
Sorry, but OCB is not for adding features. Only for bugfixed not merged in Odoo itself. Please do a module instead.

**Comment by NSKTHR** [comment]
Hi again, 
How to test this? I mean if it were a module i can install and then see if it does the work or not.

thanks

**Comment by rrakso** [comment]
@pedrobaeza Ok I get it, can someone help me with module making for this "feature"? I'm newbie in Odoo, so little help will be great! :nerd_face:

**Comment by pedrobaeza** [comment]
Check https://odoo-community.org/shop/product/stock-mts-mto-rule-2466

**Comment by rrakso** [comment]
@NSKTHR just replace that file ``addons/stock/models/stock_rule.py`` (in your Odoo)  with [this](https://github.com/OCA/OCB/blob/46cd5d274c69accb4a9e2f77253ca9f1630560ba/addons/stock/models/stock_rule.py) one, and then 
image
(hit upgrade button), after that you can choose new supply method 
image 
this new method help with [issue](https://github.com/OCA/stock-logistics-warehouse/issues/727#issuecomment-570613107) in Odoo 13

**Comment by rrakso** [comment]
> Check https://odoo-community.org/shop/product/stock-mts-mto-rule-2466

@pedrobaeza Yup! This is it! We were using it in Odoo 12, but after upgrade this module was no longer working for us, so I've created this "fix" (?), it adds new method to Odoo 13 which work the same as [Stock MTS+MTO Rule](https://odoo-community.org/shop/product/stock-mts-mto-rule-2466)
This module doesn't support Odoo 13, look
image
People were talking about it earlier [here](https://github.com/OCA/stock-logistics-warehouse/issues/727#issuecomment-570244101)

**Comment by NSKTHR** [comment]
Hi, thanks a lot for your reply.

Where do you want me to change the supply method?
There are like 8 rules in my odoo and 5 Routes. I have chosen "Replenish on Order (MTO)" and "Manufacture" as routes for my products.

On which rule should i change it to MTS+MTO by ERS? ( replenish on order is applied on 2 rules, on one the supply method is set to "take from stock, if unavailable , trigger another rule" and on the other it is set to "Trigger another rule" )

**Comment by rrakso** [comment]
@NSKTHR we have also the "Replenish on Order (MTO)", look:
image
We have made the change in the **first rule** (that is, from _STOCK_ to _OUTPUT_)
image
The other down **is not changed**, in our case, that is:
image
## Important
Today we've made some tests, and my previous code had one bug, I fixed it today, now (after some tests) it seems, that everything is working correctly. [Here](https://gitlab.com/rakso/odoo/blob/5708c2a3628c83515ba75db64f969c5527be8741/addons/stock/models/stock_rule.py) is the updated file, which need to be changed. 

Let me know, if you can handle it! If something, then I can always help! :smile:

**Comment by NSKTHR** [comment]
I get this error:


Error:
Odoo Server Error

Traceback (most recent call last):
  File "C:\Programs\Odoo 13.0\server\odoo\http.py", line 619, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "C:\Programs\Odoo 13.0\server\odoo\http.py", line 309, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "C:\Programs\Odoo 13.0\server\odoo\tools\pycompat.py", line 14, in reraise
    raise value
  File "C:\Programs\Odoo 13.0\server\odoo\http.py", line 664, in dispatch
    result = self._call_function(**self.params)
  File "C:\Programs\Odoo 13.0\server\odoo\http.py", line 345, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "C:\Programs\Odoo 13.0\server\odoo\service\model.py", line 93, in wrapper
    return f(dbname, *args, **kwargs)
  File "C:\Programs\Odoo 13.0\server\odoo\http.py", line 338, in checked_call
    result = self.endpoint(*a, **kw)
  File "C:\Programs\Odoo 13.0\server\odoo\http.py", line 909, in __call__
    return self.method(*args, **kw)
  File "C:\Programs\Odoo 13.0\server\odoo\http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "C:\Programs\Odoo 13.0\server\odoo\addons\web\controllers\main.py", line 1323, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "C:\Programs\Odoo 13.0\server\odoo\addons\web\controllers\main.py", line 1311, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "C:\Programs\Odoo 13.0\server\odoo\api.py", line 411, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "C:\Programs\Odoo 13.0\server\odoo\api.py", line 398, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "C:\Programs\Odoo 13.0\server\odoo\addons\sale_management\models\sale_order.py", line 122, in action_confirm
    res = super(SaleOrder, self).action_confirm()
  File "C:\Programs\Odoo 13.0\server\odoo\addons\sale\models\sale.py", line 742, in action_confirm
    self._action_confirm()
  File "C:\Programs\Odoo 13.0\server\odoo\addons\sale_stock\models\sale_order.py", line 92, in _action_confirm
    self.order_line._action_launch_stock_rule()
  File "C:\Programs\Odoo 13.0\server\odoo\addons\procurement_jit\sale.py", line 11, in _action_launch_stock_rule
    res = super(SaleOrderLine, self)._action_launch_stock_rule(previous_product_uom_qty=previous_product_uom_qty)
  File "C:\Programs\Odoo 13.0\server\odoo\addons\sale_stock\models\sale_order.py", line 498, in _action_launch_stock_rule
    self.env['procurement.group'].run(procurements)
  File "C:\Programs\Odoo 13.0\server\odoo\addons\mrp\models\stock_rule.py", line 137, in run
    return super(ProcurementGroup, self).run(procurements_without_kit)
  File "C:\Programs\Odoo 13.0\server\odoo\addons\stock\models\stock_rule.py", line 430, in run
    raise UserError('\n'.join(errors))
  File "C:\Programs\Odoo 13.0\server\odoo\addons\stock\models\stock_rule.py", line 228, in _run_pull
    if rule.procure_method == 'mts_else_mto' or rule.procure_method == 'mts_else_mto_ByERS':
AttributeError: 'stock.rule' object has no attribute 'produce_method'

**Comment by NSKTHR** [comment]
@rrakso Please let me know if some other file has to be replaced too

**Pull request closed without merging**